database/migrations/2019_09_20_112334_create_journal_entries_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateJournalEntriesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('journal_entries', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->date('written_at')->unique();
+            $table->text('written_at_raw');
```

Comment:
any reason to use `TEXT` here?

```suggestion
            $table->string('written_at_raw');
```

(same bellow for `weather` and `weather_raw`)

Reply:
The only reason was that I wanted to make our lives easier by not having to set the column size. Happy to change it. But I'm curious as to why? Is it that we can't set an index on a `TEXT` column?

Reply:
https://qr.ae/TWsnZt

> For most types, all the data for each row is stored together. The larger types: blob and text, are stored separately. This means that there is a performance impact.